### PR TITLE
Adds Honeycomb

### PR DESCRIPTION
**Pull request by StetsenTech**
**Ticket(s)**:
https://app.asana.com/0/1133543009734854/1153847903754889/f

**Summary**:
- Adds Honeycomb

### CODE REVIEW

**Comment by StetsenTech** [comment]
I think I pushed the `Gemfile.lock` at the same time you made the comment. LOL!

File: config/initializers/honeycomb.rb [hunk 1]
```diff
@@ -0,0 +1,4 @@
+Honeycomb.configure do |config|
+  config.write_key = Rails.application.secrets.honeycomb_api_key
```
Comment (sadleb) [inline config/initializers/honeycomb.rb]: I don't think Honeycomb should be required. E.g. it's fine to have an ENV with no `HONEYCOMB_DATASET` or `HONEYCOMB_API_KEY` -- does this fail gracefully if they aren't set?

Also, given that we'll want to be able to have an env that doesn't use Honeycomb, we should set `require=false` in the Gemfile, check if the ENV vars are set here, and explicitly require the gem if so. That way, if we have a Review App or some other environment where we aren't using Honeycomb, the gem won't load into memory on restart making it faster and using less memory (e.g. if we want to use cheap or free dynos to run the app which come with very little memory).

File: config/secrets.yml [hunk 2]
```diff
@@ -6,6 +6,8 @@ development:
   mailer_from_email: <%= ENV['MAILER_FROM_EMAIL'] %>
   devise_secret_key: <%= ENV['DEVISE_SECRET_KEY'] %>
   devise_pepper: <%= ENV['DEVISE_PEPPER'] %>
+  honeycomb_dataset: <%= ENV['HONEYCOMB_DATASET'] %>
+  honeycomb_api_key: <%= ENV['HONEYCOMB_API_KEY'] %>
```
Comment (sadleb) [inline config/secrets.yml]: For Portal on Heroku, I used this: https://github.com/honeycombio/heroku-honeycomb-drain -- the ENV key is name `HONEYCOMB_WRITE_KEY`. Can we name it the same here for consistency, esp if we decide to use another instance of this honeycomb drain app for platform?

**Comment by sadleb** [comment]
Here is a task with more info on how I setup the honeycomb drain for Portal: https://app.asana.com/0/search/1154069427979331/1147796411992721

File: config/initializers/honeycomb.rb (hunk 1 above)
Comment (rshipp) [reply config/initializers/honeycomb.rb]: good point! i hadnt thought of that

Comment (StetsenTech) [reply config/initializers/honeycomb.rb]: I tested on my local env with nothing present before pushing and everything still functioned properly.

I am more than happy to set `require=false`

File: config/secrets.yml (hunk 2 above)
Comment (StetsenTech) [reply config/secrets.yml]: Done!
https://github.com/bebraven/platform/pull/51/files

File: config/initializers/honeycomb.rb (hunk 1 above)
Comment (StetsenTech) [reply config/initializers/honeycomb.rb]: Done!
https://github.com/bebraven/platform/pull/51/files

**Pull request merged**